eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -0,0 +1,486 @@
+
+  val RECONNECT_TIMER = "reconnect"
+
+  val SEND_PING_TIMER = "send_ping"
+
+  val MAX_FUNDING_TX_ALREADY_SPENT = 10
+
+  val MAX_FUNDING_TX_NOT_FOUND = 10
+
+  val IGNORE_NETWORK_ANNOUNCEMENTS_PERIOD = 5 minutes
+
+  def props(nodeParams: NodeParams, switchboard: ActorRef, router: ActorRef) = Props(new PeerConnection(nodeParams, switchboard, router))
+
+  // @formatter:off
+
+  sealed trait Data
+  sealed trait HasTransport extends Data { def transport: ActorRef }
+  case class Nothing() extends Data
+  case class AuthenticatingData(pendingAuth: PendingAuth, transport: ActorRef) extends Data
+  case class BeforeInitData(remoteNodeId: PublicKey, pendingAuth: PendingAuth, transport: ActorRef) extends Data with HasTransport
+  case class InitializingData(nodeParams: NodeParams, pendingAuth: PendingAuth, remoteNodeId: PublicKey, transport: ActorRef, peer: ActorRef, localInit: wire.Init) extends Data with HasTransport
+  case class ConnectedData(nodeParams: NodeParams, remoteNodeId: PublicKey, transport: ActorRef, peer: ActorRef, localInit: wire.Init, remoteInit: wire.Init, rebroadcastDelay: FiniteDuration, gossipTimestampFilter: Option[GossipTimestampFilter] = None, behavior: Behavior = Behavior(), expectedPong_opt: Option[ExpectedPong] = None) extends Data with HasTransport
+  case class ExpectedPong(ping: Ping, timestamp: Long = Platform.currentTime)
+  case class PingTimeout(ping: Ping)
+
+  sealed trait State
+  case object BEFORE_AUTH extends State
+  case object AUTHENTICATING extends State
+  case object BEFORE_INIT extends State
+  case object INITIALIZING extends State
+  case object CONNECTED extends State
+
+  case class InitializeConnection(peer: ActorRef)
+  case object ResumeAnnouncements
+  case object SendPing
+
+  case class PendingAuth(connection: ActorRef, remoteNodeId_opt: Option[PublicKey], address: InetSocketAddress, origin_opt: Option[ActorRef], transport_opt: Option[ActorRef] = None) {
+    def outgoing: Boolean = remoteNodeId_opt.isDefined // if this is an outgoing connection, we know the node id in advance
+  }
+  case class Authenticated(peerConnection: ActorRef, remoteNodeId: PublicKey, address: InetSocketAddress, outgoing: Boolean, origin_opt: Option[ActorRef])
+  case class AuthenticationFailed(address: InetSocketAddress) extends RuntimeException(s"connection failed to $address")
+  case class ConnectionReady(peerConnection: ActorRef, remoteNodeId: PublicKey, address: InetSocketAddress, outgoing: Boolean, localInit: wire.Init, remoteInit: wire.Init)
+
+  case class DelayedRebroadcast(rebroadcast: Rebroadcast)
+
+  sealed trait BadMessage
+  case class InvalidSignature(r: RoutingMessage) extends BadMessage
+  case class InvalidAnnouncement(c: ChannelAnnouncement) extends BadMessage
+  case class ChannelClosed(c: ChannelAnnouncement) extends BadMessage
+
+  case class Behavior(fundingTxAlreadySpentCount: Int = 0, fundingTxNotFoundCount: Int = 0, ignoreNetworkAnnouncement: Boolean = false)
```

Comment:
`fundingTxNotFoundCount` seems unused, remove?

Reply:
Good catch. This is handled in the `InvalidAnnouncement` case, where we disconnect immediately. We could keep the count because disconnection is asynchronous, but it seems useless.